cdap-standalone/bin/cdap.sh

```diff
@@ -100,7 +100,7 @@ fi
 
 # java version check
 JAVA_VERSION=`$JAVACMD -version 2>&1 | grep "java version" | awk '{print $3}' | awk -F '.' '{print $2}'`
-if [ $JAVA_VERSION -ne 7 ] && [ $JAVA_VERSION -ne 8 ]; then
+if [ ! -z $JAVA_VERSION ] && [ $JAVA_VERSION -ne 7 ] && [ $JAVA_VERSION -ne 8 ]; then
```

Comment:
Well, this is kinda funny... On my new Mac, I'd not installed Java, yet. However, there's a `java` binary, which is a symlink to xcode-select (or something like that) which prompts to install Java. However, this tool exits 0, so we run the tool, it produces no output (it presents a popup in the UI) and `JAVA_VERSION` ends up set, but empty. Using `-n` would falsely continue, and we get a complaint from Bash about `unary operator expected` and the entire `if` is skipped.